server/src/main/scala/org/broadinstitute/dsp/workbench/welder/server/ObjectService.scala

```diff
@@ -180,18 +193,18 @@ class ObjectService(
           }
     }
 
-  private def findStorageLink(localPath: java.nio.file.Path): IO[Option[(Boolean, StorageLink)]] = for {
-    storageLinks <- storageLinksCache.get
-    baseDirectory <- IO.fromEither(getLocalBaseDirectory(localPath).leftMap(s => BadRequestException(s)))
-  } yield {
-    storageLinks.find(x => x._1.path == baseDirectory).map {
+  private def findStorageLink[A](localPath: java.nio.file.Path, safeModeResponse: StorageLink => Kleisli[IO, TraceId, A], editModeResponse: StorageLink => Kleisli[IO, TraceId, A]): Kleisli[IO, TraceId, A] = for {
+    storageLinks <- Kleisli.liftF(storageLinksCache.get)
+    baseDirectory <- Kleisli.liftF(IO.fromEither(getLocalBaseDirectory(localPath).leftMap(s => BadRequestException(s))))
+    storagelink = storageLinks.find(x => x._1.path == baseDirectory)
```

Comment:
What is `path` here? Is this a real variable? Shouldn't this be comparing against the localBaseDirectoryPath?

FWIW I can't seem to produce a storageLink hit locally - I tried patching this and still see the same issue:

```
curl localhost:8080/storageLinks                                                                   
{"storageLinks":[{"localBaseDirectory":"foo","localSafeModeBaseDirectory":"foo-safe","cloudStorageDirectory":"gs://fc-0bf6534f-a9cb-4d58-b311-c6a1c65a359f/welder-test","pattern":""}]}

curl -X POST localhost:8080/objects/metadata -d '{"localPath": "foo/bar.ipynb"}'
{"errorMessage":"No storage link found for foo/bar.ipynb","errorCode":1}
```

Reply:
[this](https://github.com/DataBiosphere/welder/blob/master/core/src/main/scala/org/broadinstitute/dsp/workbench/welder/model.scala#L62) is how baseDirectory is defined. In storagelinks, we store both `localBaseDirectory`, and `localSafeModeBaseDirectory`, and when we find a match, we then determine which one it is based on its type.

> What is path here? Is this a real variable? Shouldn't this be comparing against the localBaseDirectoryPath?

Not sure what you mean by "real variable". Its type is java.nio.file.Path, and it can be either `localBaseDirectoryPath` or `localSafeBaseDirectoryPath`. We're comparing it against calculated "baseDirectory" (which is also java,nio.file.Path) from "localPath" in user request.

Reply:
OK, I can't fully trace the caching + `LocalDirectory` so I'll assume that part works.

This line I think is a problem:
https://github.com/DataBiosphere/welder/blob/70d3d77060c772af1510170f9ee2312e3fe326e0/core/src/main/scala/org/broadinstitute/dsp/workbench/welder/package.scala#L46

My example requests I gave of "/workspaces/ws1" were just an example of how the applications might structure the directories. Welder **should not make any assumptions about these path structures or any assumptions about workspaces**. It should just accept any base directory path. All paths should be relative to the Jupyter user home directory (same as the Welder volume).